src/gluonts/dataset/artificial/_base.py

```diff
@@ -513,34 +516,40 @@ def sigmoid(x: np.ndarray) -> np.ndarray:
             length = state.randint(low=self.length_low, high=self.length_high)
             start = self._get_start(i, my_random)
             envelope = sigmoid((np.arange(length) - 20.0) / 10.0)
-            if self.is_scale:
-                scale = 0.1 * val_range * state.random_sample()
             level = 0.3 * val_range * (state.random_sample() - 0.5)
             phi = 2 * np.pi * state.random_sample()
-            T = self._get_period()
-            w = 2 * np.pi / T
-            D = 0.02 * val_range * state.random_sample()
-            if self.is_noise:
-                noise = D * state.normal(size=length)
+            period = self._get_period()
+            w = 2 * np.pi / period
             t = np.arange(length)
             idx = pd.date_range(
                 start=start, freq=self.freq_str, periods=length
             )
             special_tp_indicator = self._special_time_point_indicator(idx)
             sunday_effect = state.random_sample() * special_tp_indicator
             v = np.sin(w * t + phi) + sunday_effect
+
             if self.is_scale:
+                scale = 0.1 * val_range * state.random_sample()
                 v *= scale
             v += level
             if self.is_noise:
+                noise_range = 0.02 * val_range * state.random_sample()
+                noise = noise_range * state.normal(size=length)
                 v += noise
             v = envelope * v
-            z = np.zeros_like(v)
-            v = np.max(np.vstack([v, z + self.min_val]), axis=0)
-            v = np.min(np.vstack([v, z + self.max_val]), axis=0)
+            if self.clip_values:
+                np.clip(v, a_min=self.min_val, a_max=self.max_val, out=v)
+            else:
+                v_min = v.min()
+                v_max = v.max()
+                p_min = max(self.min_val, v_min)
+                p_max = min(self.max_val, v_max)
+                desired_min = np.clip(p_min + (p_max - v_max), a_min=self.min_val, a_max=self.max_val)
+                desired_max = np.clip(p_max + (p_min - v_min), a_min=self.min_val, a_max=self.max_val)
```

Comment:
I don't understand the logic here: scaling each time series so that it lies between `self.min_val` and `self.max_val` should be simpler than this, so I'm assuming that something different is intended to happen here?

What are `desired_min` and `desired_max` supposed to be? Are you trying to leave the lower bound of time series not going below `self.min_val` untouched, and the upper bound of time series not going below `self.max_val` also untouched? I'm just guessing

Reply:
Yes rescaling to `[self.min_val, self.max_val]` is simpler but then all the time series have the same min and max values. From what I understood self.min_val and self.max_val are bounds in which all the time series should be but are not necessarily reached.

So the idea was to keep the same length (v_max - v_min) but to put push values in [self.min_val, self.max_val]. E.g. with self.min_val = 0, self.max_val = 10, v_min = -10, v_max = -4, we want to keep the range's length 6 and we get desired_min = 0, desired_max = 6.

Then with desired_min, desired_max it's just an affine mapping.

Let me know if it's not clear or if there's an easier way to do it.

Reply:
Maybe you can add some explanatory comments about the `desired_min` and `desired_max` and/or change the name to a more descriptive one such as `rescaled_min`, `shifted_min` or something similar.